lib/hooks/update-action/index.js

```diff
@@ -0,0 +1,47 @@
+const { get } = require('lodash');
+const { NotFoundError, BadRequestError } = require('@asl/service/errors');
+
+module.exports = settings => {
+
+  const { Project, ProjectVersion, Profile, Establishment } = settings.models;
+
+  return async model => {
+    const type = get(model, 'data.model');
+    const action = get(model, 'data.action');
+    const id = get(model, 'data.id');
+
+    if (type === 'project' && action === 'grant') {
+      const project = await Project.query().findById(id);
+      const version = await ProjectVersion.query().orderBy('createdAt', 'desc').findOne({ projectId: id });
+      const transferToEstablishmentId = get(version, 'data.transferToEstablishment');
+
+      // not a transfer - continue.
+      if (!transferToEstablishmentId) {
+        return Promise.resolve();
+      }
+
+      const transferToEstablishment = await Establishment.query().findById(transferToEstablishmentId);
+
+      if (!transferToEstablishment) {
+        throw new NotFoundError();
```

Comment:
What happens to the task or project/version if one of these errors is thrown?

Reply:
god knows

Reply:
It works fine :-)

Reply:
clever girl